waspc/test/Generator/FileDraft/CopyAndModifyTextFileDraftTest.hs

```diff
@@ -0,0 +1,29 @@
+module Generator.FileDraft.CopyAndModifyTextFileDraftTest where
+
+import Fixtures (systemSPRoot)
+import qualified Generator.MockWriteableMonad as Mock
+import qualified StrongPath as SP
+import Test.Tasty.Hspec
+import Wasp.Generator.FileDraft
+
+spec_CopyAndModifyTextFileDraft :: Spec
+spec_CopyAndModifyTextFileDraft = do
+  describe "write" $ do
+    it "Creates new file by copying existing text file and applying modifications to it" $ do
+      let mock = write dstDir fileDraft
+      let mockLogs = Mock.getMockLogs mock Mock.defaultMockConfig
+      Mock.createDirectoryIfMissing_calls mockLogs
+        `shouldBe` [(True, SP.toFilePath $ SP.parent expectedDstPath)]
+      Mock.readFileAsText_calls mockLogs
+        `shouldBe` [SP.toFilePath expectedSrcPath]
+      Mock.writeFileFromText_calls mockLogs
+        `shouldBe` [(SP.toFilePath expectedDstPath, "First line\nMock text file content")]
+  where
+    (dstDir, dstPath, srcPath) =
```

Comment:
Is this a way in Haskell to group variables? (deconstructing a tuple on the left while defining it on the lef)

Reply:
Well not officially, I just used it to group them :D, but I actually didn't achieve much here so now I put them all into separate lines.